sdk/src/test/java/com/mapbox/maps/MapboxMapTest.kt

```diff
@@ -162,16 +161,60 @@ class MapboxMapTest {
 
     val styleLoadCallback = mockk<Style.OnStyleLoaded>(relaxed = true)
 
-    mapboxMap.onFinishLoadingStyleExtension(style, styleExtension, styleLoadCallback)
+    val userCallbackStyleSlots = mutableListOf<Style.OnStyleLoaded?>()
+    val callbackStyleSlots = mutableListOf<Style.OnStyleLoaded?>()
+    val callbackStyleSpritesSlots = mutableListOf<Style.OnStyleLoaded?>()
+    val callbackStyleSourcesSlots = mutableListOf<Style.OnStyleLoaded?>()
+
+    mapboxMap.loadStyle(styleExtension, styleLoadCallback)
+
+    verify {
+      styleObserver.setLoadStyleListener(
+        captureNullable(userCallbackStyleSlots),
+        captureNullable(callbackStyleSlots),
+        captureNullable(callbackStyleSpritesSlots),
+        captureNullable(callbackStyleSourcesSlots),
+        any(),
+      )
+    }
+
+    verifyNo { source.bindTo(style) }
+    verifyNo { image.bindTo(style) }
+    verifyNo { layer.bindTo(style, layerPosition) }
+    verifyNo { light.bindTo(style) }
+    verifyNo { terrain.bindTo(style) }
+    verifyNo { projection.bindTo(style) }
+    verifyNo { atmosphere.bindTo(style) }
+    verifyNo { styleLoadCallback.onStyleLoaded(style) }
+
+    callbackStyleSlots.first()!!.onStyleLoaded(style)
 
-    assertEquals(mapboxMap.style, style)
-    verify { source.bindTo(style) }
-    verify { image.bindTo(style) }
-    verify { layer.bindTo(style, layerPosition) }
     verify { light.bindTo(style) }
     verify { terrain.bindTo(style) }
     verify { projection.bindTo(style) }
     verify { atmosphere.bindTo(style) }
+    verifyNo { source.bindTo(style) }
+    verifyNo { image.bindTo(style) }
+    verifyNo { layer.bindTo(style, layerPosition) }
+    verifyNo { styleLoadCallback.onStyleLoaded(style) }
+
+    // TODO
```

Comment:
if public ticket will be in place - mention link here otherwise not clear what's the TODO